src/coreclr/jit/importervectorization.cpp

```diff
@@ -324,6 +324,11 @@ GenTree* Compiler::impCreateCompareInd(GenTreeLclVar*        obj,
     GenTree*  addOffsetTree = gtNewOperNode(GT_ADD, TYP_BYREF, obj, offsetTree);
     GenTree*  indirTree     = gtNewIndir(type, addOffsetTree);
 
+    if (varTypeIsSmall(indirTree))
+    {
+        indirTree = gtNewCastNode(actualType, indirTree, true, varTypeToUnsigned(type));
```

Comment:
This will work, but it's kind of unnecessary, morph retype the `IND` and drop the cast. The original fix with the `USHORT` indir looked good.

Reply:
But presumably it emits the same IR as other stuff in the importer, right? Changed to USHORT, thanks!

Reply:
> But presumably it emits the same IR as other stuff in the importer, right?

Well, any IR is just IR. `USHORT` indirs are perfectly fine, `CAST<ushort>(IND<short>)` are also fine, one's just smaller.